src/poll.ts

```diff
@@ -44,19 +58,28 @@ export const poll = async (options: Options): Promise<string> => {
       `Retrieved ${result.data.check_runs.length} check runs named ${checkName}`
     )
 
-    const completedCheck = result.data.check_runs.find(
+    const completedChecks = result.data.check_runs.filter(
       checkRun => checkRun.status === 'completed'
     )
-    if (completedCheck) {
-      log(
-        `Found a completed check with id ${completedCheck.id} and conclusion ${completedCheck.conclusion}`
+    if (
+      completedChecks.length &&
+      (!waitForAll || completedChecks.length === result.data.check_runs.length)
```

Comment:
Hey!

First of all thank you for this PR!

Trying to use your solution but cannot understand 2 points:
1. Why do we need `completedChecks.length` in if statement on first position? 
2. How it may be true if `result.data.check_runs.length` will be always +1 of `completedChecks.length` because of this exact waitForAll check is running and not completed?

So, for example I have 2 more check-runs + this action check-run.
Once other 2 checks passing I see `completedChecks.length` equals 2 and `result.data.check_runs.length` equals 3 which makes if statement false continuously.

Cannot understand what was the idea to make `if` this way as it keep failing if added as it is in my check-runs due to:

```
waitForAll = true
completedChecks.length = 2
result.data.check_runs.length = 3

if (2 && (false || 2 === 3)) // results to false
```
 
Thank you in advance.

Reply:
Hi,

Sorry for not getting back to you sooner.

1. I left the `completedChecks.length` in to preserve backward compatibility where if at least one check was found it would return `completedCheck.conclusion`. I didn't want to break any existing users.
2. For my use case I was waiting for a check with a different name. I have a job named `Deploy` that uses `action-wait-for-check` to wait for jobs named `Build`. I guess you could compare the current job name to the `checkName` and use `completedChecks - 1`.